src/config/machines/example-grant-machine.js

```diff
@@ -23,11 +23,41 @@ export const actionImplementations = {
     currentPageId: (_context, event) => {
       return event.type === 'NEXT' ? event.nextPageId : event.previousPageId;
     }
+  }),
+  /**
+   * Logs a message when a page is loaded
+   * @param {object} context machine context
+   */
+  loadPageAction: (context) => {
+    console.log(`${context.currentPageId} page loaded. State should reset in 1 second.`);
+  },
+
+  resetState: assign({
+    /**
+     * Resets the current page id.
+     * Sets the currentPageId to 'start' and clears the completedPageIds array.
+     * @param {object} _context - The machine context.
+     * @param {object} _event - The triggered event.
+     * @returns {string} The start page id.
+     */
+    currentPageId: (_context, _event) => {
+      return 'start';
+    },
+
+    /**
+     * Resets the completed page ids array
```

Comment:
Does this reset anything in itself?

Reply:
Yes it resets the completed pages ids?

Reply:
Okay just seems to return an empty array to me but I think I'm missing something